clients/src/test/java/org/apache/kafka/clients/NetworkClientTest.java

```diff
@@ -960,4 +1112,29 @@ public KafkaException getAndClearFailure() {
             return failure;
         }
     }
+
+    private class AddressChangeHostResolver implements HostResolver {
+        private boolean useNewAddresses;
+        private InetAddress[] initialAddressArray = initialAddresses.toArray(new InetAddress[0]);
+        private InetAddress[] newAddressArray = newAddresses.toArray(new InetAddress[0]);
```

Comment:
nit: I wonder if it would be better to pass these two in a constructor? It makes reading the tests a bit more easy.